modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/components/PageEditor.js

```diff
@@ -133,3 +125,83 @@ function Row({children, item}) {
 		rowContent
 	);
 }
+
+function Column({children, item}) {
+	// eslint-disable-next-line
+	const [{canDrop, isDragging}, drop] = useDrop({
+		accept: [LAYOUT_DATA_ITEM_TYPES.fragment]
+	});
+
+	const {size} = item.config;
+
+	return (
+		<div className={classNames('col', {[`col-${size}`]: size})} ref={drop}>
+			{children}
+		</div>
+	);
+}
+
+function Fragment({item}) {
+	const {fragmentEntryLinks} = useContext(StoreContext);
+
+	const fragmentEntryLink =
+		fragmentEntryLinks[item.config.fragmentEntryLinkId];
+
+	let markup = '';
+
+	if (typeof fragmentEntryLink.content === 'string') {
+		markup = fragmentEntryLink.content;
+	} else if (fragmentEntryLink.content.value && fragmentEntryLink.content.value.content) {
+		markup = fragmentEntryLink.content.value.content;
+	} else {
+		markup = `<div>No markup from ${item.config.fragmentEntryLinkId}</div>`;
+	}
+
+	return (
+		<UnsafeHTML
+			className="fragment"
+			markup={markup}
+		/>
+	);
+}
+
+const LAYOUT_DATA_ITEMS = {
+	[LAYOUT_DATA_ITEM_TYPES.column]: Column,
+	[LAYOUT_DATA_ITEM_TYPES.container]: Container,
+	[LAYOUT_DATA_ITEM_TYPES.fragment]: Fragment,
+	[LAYOUT_DATA_ITEM_TYPES.root]: Root,
+	[LAYOUT_DATA_ITEM_TYPES.row]: Row
+};
+
+const LayoutDataItem = React.forwardRef(({item, layoutData}, ref) => {
```

Comment:
Maybe we won't even need a `forwardRef`. Would that then be used to add the correct `drag`?

Reply:
Another good question.

Once the items have been "dropped" from the "sidebar" to the "editor" we also need to be able to drag and drop them inside the "editor", so I assumed we would need this `forwardRef` because we need to be able to pass some kind of `ref` (the second element returned by the `useDrag` hook) to the `LayoutDataItem`  component.

Reply:
Yeah, I thought I'd do everything here but it looks like this is the best place, I was assuming we could only move one Fragment and one Container so I was thinking of adding `useDrag` to their respective components. I think there might be problems with components that we don't want to move in the case of columns.